### DISCUSSION TITLE
Looping over custom simplification rules

### DISCUSSION THREAD

**Discussion by JacobBeaudry** [josdejong/mathjs#2600] [Q&A] [answered]
HI Jos!

I was trying to add some custom simplify rules as objects, with the `repeat` option (e.g. `{ l: '-(n1 + n2)', r: '-1 * (n1 + n2)', repeat: true }`); however, the `SimplifyRule` type for objects has been defined to only accept `{ l: string; r: string }` and appears to be missing `repeat?: boolean` as well as the other options. 

As you know, I'm not a programmer, so if there is a simple work-around to this, sorry for bringing it up, unnecessarily!

Thanks in advance for your help!

Jacob

**Comment by josdejong** [comment, marked as answer]
Thanks for bringing this up. So the TypeScript definition `SimplifyRule` is missing the parameters `s`, `repeat`, `assuming`, and `imposeContext`. Also, the type definitions of the options argument of `simplify` is missing `consoleDebug` and `context`.

For the time being you can probably use a `// @ts-ignore` to work around it.

Documentation: https://github.com/josdejong/mathjs/blob/develop/src/function/algebra/simplify.js#L116-L158

Anyone able to improve the TypeScript definitions?

**Reply by JacobBeaudry** [reply]
I notified our team and someone opened a PR. Thanks for the quick reply and work-around! Hope all's well on your end!

**Reply by josdejong** [reply]
The improved TypeScript definitions are now updated in `v11.0.0`
